apt-hook/json-hook-src/json-hook.go

```diff
@@ -0,0 +1,249 @@
+	Architecture string `json:"architecture"`
+	Pin          int    `json:"pin"`
+	Origins      []struct {
+		Archive  string `json:"archive"`
+		Codename string `json:"codename"`
+		Version  string `json:"version"`
+		Origin   string `json:"origin"`
+		Label    string `json:"label"`
+		Site     string `json:"site"`
+	} `json:"origins"`
+}
+type jsonRPC struct {
+	JsonRPC string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	Params  struct {
+		Command         string   `json:"command"`
+		UnknownPackages []string `json:"unknown-packages"`
+		Packages        []struct {
+			Id           int    `json:"id"`
+			Name         string `json:"name"`
+			Architecture string `json:"architecture"`
+			Mode         string `json:"mode"`
+			Automatic    bool   `json:"automatic"`
+			Versions     struct {
+				Candidate jsonRPCPackageVersion `json:"candidate"`
+				Install   jsonRPCPackageVersion `json:"install"`
+				Current   jsonRPCPackageVersion `json:"current"`
+			} `json:"versions"`
+		} `json:"packages"`
+	} `json:"params"`
+}
+
+func createUpdateMessage(standardSecurityCount int, esmInfraCount int, esmAppsCount int) string {
+	displayStandard := true
+	displayEsmInfra := true
+	displayEsmApps := true
+	if standardSecurityCount == 0 {
+		displayStandard = false
+	}
+	if esmInfraCount == 0 {
+		displayEsmInfra = false
+	}
+	if esmAppsCount == 0 {
+		displayEsmApps = false
+	}
+
+	if !displayStandard && !displayEsmInfra && !displayEsmApps {
+		return ""
+	}
+
+	esmInfraFirst := false
+	esmAppsFirst := false
+	if !displayStandard && displayEsmInfra {
+		esmInfraFirst = true
+	} else if !displayStandard && !displayEsmInfra && displayEsmApps {
+		esmAppsFirst = true
+	}
+
+	standardUpdates := ""
+	esmInfraUpdates := ""
+	esmAppsUpdates := ""
+	if displayStandard {
+		standardUpdates = fmt.Sprintf("%d standard security ", standardSecurityCount)
+		if standardSecurityCount > 1 {
+			standardUpdates += "updates"
+		} else {
+			standardUpdates += "update"
+		}
+		if displayEsmInfra && displayEsmApps {
+			standardUpdates += ","
+		}
+		if displayEsmInfra || displayEsmApps {
+			standardUpdates += " "
+		}
+		if (displayEsmInfra && !displayEsmApps) || (!displayEsmInfra && displayEsmApps) {
+			standardUpdates += "and "
+		}
+	}
+	if displayEsmInfra {
+		esmInfraUpdates = fmt.Sprintf("%d esm-infra ", esmInfraCount)
+		if esmInfraFirst {
+			esmInfraUpdates += "security "
+		}
+		if esmInfraCount > 1 {
+			esmInfraUpdates += "updates"
+		} else {
+			esmInfraUpdates += "update"
+		}
+		if displayEsmApps {
+			esmInfraUpdates += " and "
+		}
+	}
+	if displayEsmApps {
+		esmAppsUpdates = fmt.Sprintf("%d esm-apps ", esmAppsCount)
+		if esmAppsFirst {
+			esmAppsUpdates += "security "
+		}
+		if esmAppsCount > 1 {
+			esmAppsUpdates += "updates"
+		} else {
+			esmAppsUpdates += "update"
+		}
+	}
+
+	return standardUpdates + esmInfraUpdates + esmAppsUpdates
+}
+
+func fromOrigin(pkgVersion jsonRPCPackageVersion, origin string) bool {
```

Comment:
I think we can merge both of these function and always check both origin and archive just to be safe.

Reply:
@lucasmoura I'm not sure I understand what this comment is suggesting - can you describe what parameters a combined function would take and what it would do?

Reply:
@orndorffgrant Sorry for that. I mean that we can combine both `fromOrigin`  and `isSecurityUpdate`. This will allows to check both `origin` and `archive` for each security service we will provide the message.

I think ti should behave similarly to this function in update-notifier:
```python
def isSecurityUpgrade(ver):
    " check if the given version is a security update (or masks one) "
    security_pockets = [("Ubuntu", "%s-security" % DISTRO),
                        (ESM_INFRA_ORIGIN, "%s-infra-security" % DISTRO),
                        (ESM_APPS_ORIGIN, "%s-apps-security" % DISTRO),
                        ("gNewSense", "%s-security" % DISTRO),
                        ("Debian", "%s-updates" % DISTRO)]
    for (file, index) in ver.file_list:
        for origin, archive in security_pockets:
            if (file.archive == archive and file.origin == origin):
                return True
    return False
```

But we can pass the expected archive and origin directly to the function

Reply:
Ohhhh okay cool, yeah thats a good idea! I'll do this

Reply:
I added this - let me know what you think

Somewhat related: The weird thing I had to do here was get the distro codename from the package origin, which isn't ideal, but I wasn't sure of a better way to do that - any thoughts?